src/backend/base/langflow/components/vectorstores/AstraDB.py

```diff
@@ -262,8 +456,9 @@ def _build_search_args(self):
                 args["filter"] = clean_filter
         return args
 
-    def search_documents(self) -> list[Data]:
-        vector_store = self.build_vector_store()
+    def search_documents(self, vector_store=None) -> list[Data]:
+        if not vector_store:
+            vector_store = self.build_vector_store()
```

Comment:
I don't think we need this change - the check_cached decorator handles this

Reply:
So, the only reason i made this change was to more easily allow for a test to be created. The problem with the tests is that i wasnt sure how i could build the component with parameters that werent in the initial configuration - in the UI, it'll dynamically update the components based on the value of the dropdown, but is there a way to programmatically perform that same computation? i.e., something like 

`component.update_build_configuration(embedding_service = "Astra Vectorize")`

So i allowed an optional inclusion of the vector store object for purposes of the test, but that would never be used in the happy path in the component. Let me know though if you see a better way to do that